test/org/javarosa/form/api/ConstraintTextTest.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2018 Nafundi
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.javarosa.form.api;
+
+import static org.javarosa.test.utils.ResourcePathHelper.r;
+import static org.junit.Assert.assertEquals;
+;
+import org.javarosa.core.model.FormDef;
+import org.javarosa.core.test.FormParseInit;
+import org.junit.Test;
+
+public class ConstraintTextTest {
+
+    @Test
+    public void checkConstraintTexts() {
+        FormParseInit fpi = new FormParseInit(r("constraint-message-error.xml"));
+        FormEntryModel formEntryModel = fpi.getFormEntryModel();
+        FormDef formDef = fpi.getFormDef();
+        formDef.getLocalizer().setLocale("English");
+
+        formEntryModel.setQuestionIndex(formEntryModel.incrementIndex(formEntryModel.getFormIndex()));
+        FormEntryPrompt formEntryPrompt = new FormEntryPrompt(formDef, formEntryModel.getFormIndex());
+        assertEquals("Your message", formEntryPrompt.getConstraintText());
+
+        formEntryModel.setQuestionIndex(formEntryModel.incrementIndex(formEntryModel.getFormIndex()));
+        formEntryPrompt = new FormEntryPrompt(formDef, formEntryModel.getFormIndex());
+        assertEquals("Message", formEntryPrompt.getConstraintText());
+
+        formEntryModel.setQuestionIndex(formEntryModel.incrementIndex(formEntryModel.getFormIndex()));
+        formEntryPrompt = new FormEntryPrompt(formDef, formEntryModel.getFormIndex());
+        assertEquals("Your message", formEntryPrompt.getConstraintText());
+
+        formEntryModel.setQuestionIndex(formEntryModel.incrementIndex(formEntryModel.getFormIndex()));
+        formEntryPrompt = new FormEntryPrompt(formDef, formEntryModel.getFormIndex());
+        assertEquals("Message", formEntryPrompt.getConstraintText());
+    }
+}
```

Comment:
The test is a big help. Thanks. Now can we remove the duplication?

![screen shot 2018-07-04 at 9 35 51 pm](https://user-images.githubusercontent.com/55099/42302829-534fe608-7fd2-11e8-98df-dd53cd5deb63.png)

Reply:
Please add your commit if you want me to cherry pick it or let me know if you want me to do that.

Reply:
Thanks. No need to cherry pick this one.

Reply:
Fixed.